ambry-api/src/main/java/com.github.ambry/network/NetworkSend.java

```diff
@@ -59,7 +51,7 @@ public Send getPayload() {
 
   public void onSendComplete() {
     if (metrics != null) {
-      metrics.updateResponseSendTime(SystemTime.getInstance().milliseconds() - sendStartTimeInMs);
+      metrics.updateSendTime(time.milliseconds() - sendStartTimeInMs);
```

Comment:
But having said that, I am open to change the updateSendTime(long) to onSendComplete(). But then, I need to have the "Time" instance, to get the millisecs and might need some changes to the NetworkSendMetrics.